### PR TITLE
Update message template for newly published templates auto-email (Issue #9281)

### LINKED ISSUE #9281: Update message template for "new published extension"
The [message template](https://github.com/mozilla/addons-server/blob/master/src/olympia/devhub/templates/devhub/email/submission.txt) for the "new published extension" auto-email from AMO should be updated with [this text](https://docs.google.com/document/d/1msQ-vvDi7GDK04bTp8P5MJgiXoejTrSZ_EOay7HCUA0/edit#heading=h.4ln03gjbbwl3) [Google doc]. 

cc: @wagnerand

### PR DESCRIPTION
**Pull request by kvaluruk**
Fixes #9281 

Small doubt, should I add the 'Subject:' in the `submission.txt` file into `submission.html`? 
If yes, how? `<h1> ... </h1>`?

### CODE REVIEW

**Comment by kvaluruk** [comment]
@eviljeff Could you please help me with the tests? I can't figure out why most of them are failing.

File: src/olympia/devhub/templates/devhub/email/submission.txt [hunk 1]
```diff
@@ -1,27 +1,23 @@
...
 
+To learn how to help users discover your extension, stay up-to-date with news from the add-ons community, or contact the add-ons team, please visit https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/Distribution/Resources_for_publishers. 
 
-Happy developing!
+If you are interested in getting more involved with Mozilla and want to help keep the add-ons ecosystem safe and healthy, please consider becoming a volunteer add-on reviewer. You can learn more at https://wiki.mozilla.org/Add-ons/Reviewers. 
 
+Happy developing! 
 
--- 
+--
```
Comment (EnTeQuAk) [inline src/olympia/devhub/templates/devhub/email/submission.txt]: Except here! (that's the funny, not obvious part)

Email signatures have to be separated by `-- ` (dash dash whitespace) to work properly. So the whitespace at the end of the line here is actually important. (https://www.ietf.org/rfc/rfc3676.txt Section 4.3. Usenet Signature Convention)

File: src/olympia/devhub/templates/devhub/email/submission.txt [hunk 2]
```diff
@@ -1,27 +1,23 @@
-Thanks for submitting your {{ addon_name }} add-on to addons.mozilla.org (AMO)! Your add-on listing will go live shortly in our gallery here: {{ detail_url }}
+Subject: Mozilla Add-ons: {{ add-on name }} has been submitted to addons.mozilla.org! 
+
+Thanks for submitting your {{ add-on name }} add-on to addons.mozilla.org (AMO)! Your add-on listing will go live shortly in our gallery here: {{ url }}
```
[the remark this answers is not in the document]
Comment (kvaluruk) [reply src/olympia/devhub/templates/devhub/email/submission.txt]: Oh my! That was pretty stupid of me. Will make sure this won't happen the next time! :)

**Comment by kvaluruk** [comment]
@EnTeQuAk Can you please review?

File: src/olympia/devhub/templates/devhub/email/submission.txt [hunk 3]
```diff
@@ -1,23 +1,23 @@
-Subject: Mozilla Add-ons: {{ add-on name }} has been submitted to addons.mozilla.org! 
+Subject: Mozilla Add-ons: {{ addon_name }} has been submitted to addons.mozilla.org!
 
-Thanks for submitting your {{ add-on name }} add-on to addons.mozilla.org (AMO)! Your add-on listing will go live shortly in our gallery here: {{ url }}
+Thanks for submitting your {{ addon_name }} add-on to addons.mozilla.org (AMO)! Your add-on listing will go live shortly in our gallery here: {{ detail_url }}
 
 Please keep in mind that while your add-on has been screened and approved for listing, other reviewers may look into it in the future and determine that it requires changes or should be removed from the gallery. If that occurs, you will receive a separate notification with details and next steps.
 
-For more information about the review process and policies, please visit https://developer.mozilla.org/Add-ons/AMO/Policy/Reviews. 
+For more information about the review process and policies, please visit https://developer.mozilla.org/Add-ons/AMO/Policy/Reviews.
```
Comment (eviljeff) [inline src/olympia/devhub/templates/devhub/email/submission.txt]: I think we should avoid `.` at the end of urls too (either drop the `.` or always have a space, e.g. `Reviews .`) but that's more about accounting for differences in how email clients may treat the URL).

**Comment by kvaluruk** [comment]
@eviljeff Is this good?

**Comment by caitmuenster** [comment]
Thank you, @kvaluruk!!

**Pull request merged**